src/AdoNet/Shared/Storage/RelationalStorage.cs

```diff
@@ -163,11 +163,7 @@ public Task<IEnumerable<TResult>> ReadAsync<TResult>(string query, Action<IDbCom
                 throw new ArgumentNullException("selector");
             }
 
-            //It is certain the result is already ready here and can be collected straight away. Having a truly asynchronous result collection
-            //IAsyncEnumerable<TResult> or equivalent method ought to be used. Taking the result here without async-await saves for generating
-            //the async state machine.
-            var ret = ExecuteAsync(query, parameterProvider, ExecuteReaderAsync, selector, cancellationToken, commandBehavior).GetAwaiter().GetResult().Item1;
-            return Task.FromResult(ret);
+            return (await ExecuteAsync(query, parameterProvider, ExecuteReaderAsync, selector, cancellationToken, commandBehavior).ConfigureAwait(false)).Item1;
         }
```

Comment:
Are the `ConfigureAwait(false)` calls necessary? If so, why?

Reply:
From the callers perspective, this do make sense, as the context is not needed at all. Most of the times it is OK (although marginally less efficient) to not add these if all involved players do the right thing, nevertheless I remember that PostgreSQL driver was deadlocking with the Orleans execution context, and using `ConfigureAwait` in this class fixed it.

Reply:
I think that's going to be fixed in the next Postgres client release. Hopefully no other drivers cause trouble.

Reply:
Hopefully, don't know. Still it makes sense to do the right thing just in case (the right thing is using ConfigureAwait)

Reply:
@ReubenBond, @jdom It's both "the right thing, easier to reader" but also to be surer to not to have deadlocks in the operations. It's not PostgreSQL only, also MySQL can cause trouble. I think even in the latest drivers their streaming operators will deadlock.